test/hotspot/jtreg/compiler/loopopts/superword/TestDependencyOffsets.java

```diff
@@ -486,15418 +409,475 @@
+            case "vec-v016-A" -> new String[] {"-XX:MaxVectorSize=16", "-XX:+AlignVector"};
+            case "vec-v016-U" -> new String[] {"-XX:MaxVectorSize=16", "-XX:-AlignVector"};
+            case "vec-v008-A" -> new String[] {"-XX:MaxVectorSize=8", "-XX:+AlignVector"};
+            case "vec-v008-U" -> new String[] {"-XX:MaxVectorSize=8", "-XX:-AlignVector"};
+            case "vec-v004-A" -> new String[] {"-XX:MaxVectorSize=4", "-XX:+AlignVector"};
+            case "vec-v004-U" -> new String[] {"-XX:MaxVectorSize=4", "-XX:-AlignVector"};
+            default -> { throw new RuntimeException("Test argument not recognized: " + args[0]); }
+        };
+
+        CompileFramework comp = new CompileFramework();
+        long time0 = System.currentTimeMillis();
+        comp.addJavaSourceCode("InnerTest", generate(comp, flags));
+        long time1 = System.currentTimeMillis();
+        comp.compile();
+        long time2 = System.currentTimeMillis();
+        comp.invoke("InnerTest", "main", new Object[] {null});
+        long time3 = System.currentTimeMillis();
+        System.out.println("Generate: " + (time1 - time0));
+        System.out.println("Compile:  " + (time2 - time1));
+        System.out.println("Run:      " + (time3 - time2));
+    }
+
+    static record Type (String name, int size, String value, String operator, String irNode) {
+        String letter() {
+            return name.substring(0, 1).toUpperCase();
+        }
+
+        /*
+         * Template for init method generation.
+         */
+        String generateInit() {
+            return String.format("""
+                       static void init(%s[] a, %s[] b) {
+                           for (int i = 0; i < SIZE; i++) {
+                               a[i] = (%s)(2 * i);
+                               b[i] = (%s)(3 * i);
+                           }
+                       }
+                   """,
+                   name, name, name, name);
+        }
+
+        /*
+         * Template for verify method generation.
+         */
+        String generateVerify() {
+            return String.format("""
+                       static void verify(String context, %s[] aTest, %s[] bTest, %s[] aGold, %s[] bGold) {
+                           for (int i = 0; i < SIZE; i++) {
+                               if (aTest[i] != aGold[i] || bTest[i] != bGold[i]) {
+                                   throw new RuntimeException("Wrong result in " + context + " at i=" + i + ": " +
+                                                              "aTest=" + aTest[i] + ", aGold=" + aGold[i] +
+                                                              "bTest=" + bTest[i] + ", bGold=" + bGold[i]);
+                               }
+                           }
+                       }
+                   """,
+                   name, name, name, name);
+        }
+    }
+
+    static Type[] types = new Type[] {
```

Comment:
Can be made final, maybe also put in capital letters.
```suggestion
    static final Type[] TYPES = new Type[] {
```